### PR TITLE
Refactor/class args

### PR DESCRIPTION
**Pull request by rubentea16**
#88

### CODE REVIEW

**Comment by andreaschandra** [comment]
`ModuleNotFoundError: No module named 'bs4'`

**Comment by andreaschandra** [comment]
`ModuleNotFoundError: No module named 'skimage'`

**Comment by andreaschandra** [comment]
`pip install scikit-image`

**Comment by andreaschandra** [comment]
`ModuleNotFoundError: No module named 'osgeo'`
solution `conda install gdal`

**Comment by andreaschandra** [comment]
```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
/tmp/ipykernel_21402/1120933060.py in <module>
----> 1 ac = AerialCactus(root="../data")

~/earth-vision/earthvision/datasets/aerialcactus.py in __init__(self, root, train, transform, target_transform, download)
     55             self.extract_file()
     56 
---> 57         self.img_labels = self.get_path_and_label()
     58 
     59     def __getitem__(self, idx: int) -> Tuple[Any, Any]:

~/earth-vision/earthvision/datasets/aerialcactus.py in get_path_and_label(self)
     89                 self.root, 'cactus-aerial-photos', self.data_mode, self.data_mode, cat)
     90             cat_image = [os.path.join(cat_path, path)
---> 91                          for path in os.listdir(cat_path)]
     92             cat_label = [enc] * len(cat_image)
     93             image_path += cat_image

FileNotFoundError: [Errno 2] No such file or directory: '../data/cactus-aerial-photos/training_set/training_set/cactus'
``

**Comment by andreaschandra** [comment]
```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
/tmp/ipykernel_21402/271452526.py in <module>
----> 1 dd = DroneDeploy(root="../data", dataset_type='dataset-sample', data_mode = 0, 
      2             download = True)

~/earth-vision/earthvision/datasets/drone_deploy.py in __init__(self, root, dataset_type, data_mode, transform, target_transform, download)
     60 
     61         if download and not self._check_exists():
---> 62             self.download()
     63 
     64         self.load_dataset()

~/earth-vision/earthvision/datasets/drone_deploy.py in download(self)
     69         self.root = os.path.expanduser(self.root)
     70         fpath = os.path.join(self.root, self.filename)
---> 71         _urlretrieve(self.resources[self.dataset_type], fpath)
     72 
     73         if not os.path.exists(os.path.join(self.root, self.dataset_type)):

~/earth-vision/earthvision/datasets/utils.py in _urlretrieve(url, filename, chunk_size)
     22     ctx.verify_mode = ssl.CERT_NONE
     23 
---> 24     with open(filename, "wb") as fh:
     25         with urllib.request.urlopen(urllib.request.Request(url), context=ctx) as response:
     26             with tqdm(total=response.length) as pbar:

FileNotFoundError: [Errno 2] No such file or directory: '../data/dataset-sample.tar.gz'
```

**Comment by andreaschandra** [comment]
```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
/tmp/ipykernel_21402/1387867453.py in <module>
----> 1 r45 = RESISC45(root="../data", download = True)

~/earth-vision/earthvision/datasets/resisc45.py in __init__(self, root, transform, target_transform, download)
     49 
     50         if download and not self._check_exists():
---> 51             self.download()
     52             self.extract_file()
     53 

~/earth-vision/earthvision/datasets/resisc45.py in download(self)
    101         """Download and extract file."""
    102         file_url = posixpath.join(self.mirrors, self.resources)
--> 103         _urlretrieve(file_url, os.path.join(self.root, self.resources))
    104 
    105     def extract_file(self) -> None:

~/earth-vision/earthvision/datasets/utils.py in _urlretrieve(url, filename, chunk_size)
     22     ctx.verify_mode = ssl.CERT_NONE
     23 
---> 24     with open(filename, "wb") as fh:
     25         with urllib.request.urlopen(urllib.request.Request(url), context=ctx) as response:
     26             with tqdm(total=response.length) as pbar:

FileNotFoundError: [Errno 2] No such file or directory: '../data/NWPU-RESISC45.zip'
```

**Comment by andreaschandra** [comment]
please check if the root path is exists as well as other classes @rubentea16

File: earthvision/datasets/aerialcactus.py [hunk 1]
```diff
@@ -1,66 +1,89 @@
...
+        return len(self.img_labels)
 
     def get_path_and_label(self):
         """Return dataframe type consist of image path and corresponding label."""
         classes = {'cactus': 1, 'no_cactus': 0}
-        image_path = []
-        label = []
+        image_path, label = [], []
+
         for cat, enc in classes.items():
```
[the remark this answers is not in the document]
Comment (rubentea16) [reply earthvision/datasets/aerialcactus.py:87]: `enc` is used. check this https://github.com/jakartaresearch/earth-vision/blob/refactor/class-args/earthvision/datasets/aerialcactus.py#L92

**Comment by andreaschandra** [comment]
@rubentea16 is it done?

**Comment by rubentea16** [comment]
> ```
> ---------------------------------------------------------------------------
> FileNotFoundError                         Traceback (most recent call last)
> /tmp/ipykernel_21402/1120933060.py in <module>
> ----> 1 ac = AerialCactus(root="../data")
> 
> ~/earth-vision/earthvision/datasets/aerialcactus.py in __init__(self, root, train, transform, target_transform, download)
>      55             self.extract_file()
>      56 
> ---> 57         self.img_labels = self.get_path_and_label()
>      58 
>      59     def __getitem__(self, idx: int) -> Tuple[Any, Any]:
> 
> ~/earth-vision/earthvision/datasets/aerialcactus.py in get_path_and_label(self)
>      89                 self.root, 'cactus-aerial-photos', self.data_mode, self.data_mode, cat)
>      90             cat_image = [os.path.join(cat_path, path)
> ---> 91                          for path in os.listdir(cat_path)]
>      92             cat_label = [enc] * len(cat_image)
>      93             image_path += cat_image
> 
> FileNotFoundError: [Errno 2] No such file or directory: '../data/cactus-aerial-photos/training_set/training_set/cactus'
> ```

coba test lagi dong ini, di gw ga error dan kebikin folder nya @andreaschandra

**Review by andreaschandra — APPROVED** [review]

**Pull request merged**